tarantool_test.go

```diff
@@ -3148,6 +3156,387 @@ func TestConnectionFeatureOptsImmutable(t *testing.T) {
 	require.True(t, connected, "Reconnect success")
 }
 
+func TestConnection_NewWatcher(t *testing.T) {
+	test_helpers.SkipIfWatchersUnsupported(t)
+
+	const key = "TestConnection_NewWatcher"
+	connOpts := opts.Clone()
+	connOpts.RequiredProtocolInfo.Features = []ProtocolFeature{
+		WatchersFeature,
+	}
+	conn := test_helpers.ConnectWithValidation(t, server, connOpts)
+	defer conn.Close()
+
+	events := make(chan WatchEvent)
+	defer close(events)
+
+	watcher, err := conn.NewWatcher(key, func(event WatchEvent) {
+		events <- event
+	})
+	if err != nil {
+		t.Fatalf("Failed to create a watch: %s", err)
+	}
+	defer watcher.Unregister()
+
+	select {
+	case event := <-events:
+		if event.Conn != conn {
+			t.Errorf("Unexpected event connection: %v", event.Conn)
+		}
+		if event.Key != key {
+			t.Errorf("Unexpected event key: %s", event.Key)
+		}
+		if event.Value != nil {
+			t.Errorf("Unexpected event value: %v", event.Value)
+		}
+	case <-time.After(time.Second):
+		t.Fatalf("Failed to get watch event.")
+	}
+}
+
+func TestConnection_NewWatcher_unsupported(t *testing.T) {
+	test_helpers.SkipIfWatchersUnsupported(t)
```

Comment:
This check is excessive now after we have changed the behavior.

Reply:
Thank you, removed.